config/main.py

```diff
@@ -1983,8 +1983,23 @@ def load_minigraph(db, no_service_restart, traffic_shift_away, override_config,
                         fg='magenta')
             raise click.Abort()
 
-        # Dependency check golden config json
+        cfgdb_clients = db.cfgdb_clients
+        # Pick the host config in case of multi-asic
+        config_db = cfgdb_clients[DEFAULT_NAMESPACE]
+
+        cm = None
+        try:
+            cm = ConfigMgmt(configdb=config_db)
+            cm.validateConfigData()
+        except Exception as ex:
+            log.log_warning("The existing config fails YANG validation. Alerting: {}".format(ex))
+
         config_to_check = read_json_file(golden_config_path)
```

Comment:
>golden_config_path [](http://example.com/codeflow?start=41&length=18)

what if golden config is multiasic flavor? Like 

[
  {
    "localhost": {
      ...
    }
  },
  {
    "asic0": {
      ...
    }
  },
  {
    "asic1": {
      ...
    }
  }
]

Reply:
Recommend validate function without cfgdb_clients.

if multi_asic.is_multi_asic():
            for asic in multi_asic.get_namespace_list():
                target_config = config_to_check.pop(asic)
                if not SonicYangCfgDbGenerator().validate_config_db_json(target_config):
                    return False

Reply:
I want to skip Golden Config for mutliaisc right now. It will not report error if multiasic flavor.
The reason is that we have 100% table covered in single asic, and I have monitored for a while and single asic Golden Config is always YANG validated. So we have confidence it will not exit early which block the PROD flow.
But we don't have such confidence in multiasic.

Reply:
Do we have ADO track the multiasic change?

Reply:
[Task 29182008](https://msazure.visualstudio.com/One/_workitems/edit/29182008): Fail the Golden Config validation early in T2 scenario